src/modules/UI/scenes/AddToken/action.js

```diff
@@ -4,39 +4,87 @@ export const ADD_TOKEN_SUCCESS = 'ADD_TOKEN_SUCCESS'
 export const SET_TOKEN_SETTINGS = 'SET_TOKEN_SETTINGS'
 
 import {Actions} from 'react-native-router-flux'
+import * as UTILS from '../../../utils.js'
 
-import * as WALLET_API from '../../../Core/Wallets/api.js'
 import * as SETTINGS_ACTIONS from '../Settings/action.js'
+import {setCustomTokens} from '../../Settings/action.js'
 import * as SETTINGS_API from '../../../Core/Account/settings.js'
 import * as CORE_SELECTORS from '../../../Core/selectors.js'
 import * as UI_ACTIONS from '../../Wallets/action.js'
+import * as UI_WALLET_SELECTORS from '../../selectors.js'
 
+// adding a token should only be done on the account level, while enabling will occur on the wallet level
 export const addToken = (walletId, tokenObj) => (dispatch, getState) => {
   const state = getState()
+  // create modified object structure to match metaTokens
```

Comment:
This needs a major refactor and much of this action's logic should be in the reducer. Please clean up after this PR is merged.